Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdaterStrategyResolver.cs

```diff
@@ -45,89 +47,118 @@ public StrategyType GetFallbackStrategy(StrategyType strategyType)
+
+                    if (installedVersionId == latestVersionId)
+                    {
+                        _logger.LogDebug("Installed version is the same as the latest version. Using empty strategy.");
+
+                        return StrategyType.Empty;
+                    }
+
+                    if (installedVersionId < latestVersionId)
+                    {
+                        _logger.LogDebug("Installed verison is older than latest version...");
 
-                if (installedVersionId < latestVersionId)
-                {
 #if PATCHKIT_DONT_USE_DIFF_UPDATES
-                    DebugLogger.Log(
+                    _logger.LogDebug(
                         "Installed version is older than the latest version. " +
                         "Using content strategy due to define PATCHKIT_DONT_USE_DIFF_UPDATES");
-                    
+
                     return StrategyType.Content;
 #else
-                    DebugLogger.Log(
-                        "Installed version is older than the latest version. Checking whether cost of updating with diff is lower than cost of updating with content...");
-
-                    if (context.Configuration.CheckConsistencyBeforeDiffUpdate)
-                    {
-                        DebugLogger.Log("Checking consitency before allowing diff update...");
+                        int lowestVersionWithDiffId = GetLowestVersionWithDiffId(context);
+                        _logger.LogTrace("lowestVersionWithDiffId = " + lowestVersionWithDiffId);
 
-                        if (!IsVersionIntegral(context))
+                        if (installedVersionId + 1 < lowestVersionWithDiffId)
                         {
+                            _logger.LogDebug(
+                                "Diffs for updating from installed to latest version are not available. Using content strategy.");
+
                             return StrategyType.Content;
                         }
-                    }
 
-                    ulong contentCost = (ulong)GetContentCost(context); // bytes
-                    ulong diffCost = GetDiffCost(context); // bytes
-                    DebugLogger.LogVariable(contentCost, "contentCost");
-                    DebugLogger.LogVariable(diffCost, "diffCost");
-                    DebugLogger.Log(string.Format("Cost of updating with diff equals {0}.", diffCost));
-                    DebugLogger.Log(string.Format("Cost of updating with content equals {0}.", contentCost));
+                        _logger.LogDebug(
+                            "Checking whether cost of updating with diff is lower than cost of updating with content...");
 
-                    if (diffCost < contentCost)
-                    {
-                        DebugLogger.Log(
-                            "Cost of updating with diff is lower than cost of updating with content. Using diff strategy.");
+                        long contentSize = GetLatestVersionContentSize(context);
```

Comment:
Why are content size and diff size types mismatched? Shouldn't they both be `ulong`?